custom/enikshay/management/commands/drug_resistance_reconciliation.py

```diff
@@ -0,0 +1,233 @@
+        # possible values are resistant/ sensitive/ unknown
+        for drug_resistance_case in drug_resistance_cases:
+            sensitivity = drug_resistance_case.get_case_property('sensitivity')
+            if sensitivity == 'resistant':
+                resistant_drug_resistance_cases.append(drug_resistance_case)
+            elif sensitivity == 'sensitive':
+                sensitive_drug_resistance_cases.append(drug_resistance_case)
+            else:
+                if sensitivity not in self.sensitivity_values:
+                    self.sensitivity_values.append(sensitivity)
+                    print("New sensitivity value found, %s" % sensitivity)
+        resistant_drug_resistance_cases_count = len(resistant_drug_resistance_cases)
+        sensitive_drug_resistance_cases_count = len(resistant_drug_resistance_cases)
+
+        # if there is more than one case with sensitivity resistant, keep the one
+        # opened first.
+        if resistant_drug_resistance_cases_count > 1:
+            retain_case = sorted(resistant_drug_resistance_cases, key=lambda x: x.opened_on)[0]
+            self.close_cases(drug_resistance_cases, occurrence_case_id, drug_id, retain_case,
+                             "More than one resistance drug cases, picked first opened.")
+        # if there is only one case with sensitivity resistant, keep that, close rest.
+        elif resistant_drug_resistance_cases_count == 1:
+            retain_case = resistant_drug_resistance_cases[0]
+            self.close_cases(drug_resistance_cases, occurrence_case_id, drug_id, retain_case,
+                             "Only one resistance drug case found.")
+        # if there are more than one cases with sensitivity sensitive, keep the one
+        # opened first.
+        elif sensitive_drug_resistance_cases_count > 1:
+            retain_case = sorted(sensitive_drug_resistance_cases, key=lambda x: x.opened_on)[0]
+            self.close_cases(drug_resistance_cases, occurrence_case_id, drug_id, retain_case,
+                             "More than one sensitive drug cases, picked first opened.")
+        # if there is only one case with sensitivity sensitive, keep that, close rest.
+        elif sensitive_drug_resistance_cases_count == 1:
+            retain_case = sensitive_drug_resistance_cases[0].get
+            self.close_cases(drug_resistance_cases, occurrence_case_id, drug_id, retain_case,
+                             "Only one sensitive drug case found.")
+        # No case has sensitivity resistant and sensitive. Probably multiple cases with unknown status
+        # keep the one opened first, close rest.
+        else:
+            retain_case = sorted(drug_resistance_cases, key=lambda x: x.opened_on)[0]
+            self.close_cases(drug_resistance_cases, occurrence_case_id, drug_id, retain_case,
+                             "Picked first opened.")
+
+    @staticmethod
+    def _get_open_occurrence_case_ids_to_process():
```

Comment:
hmm, that would do the same thing but there are couple of advantages here
1. This would currently get ids per db shard thus reducing the number
2. The list would be shorted per shard and hence the memory footprint

I could possibly try to alter the `get_open_case_ids_in_domain_by_type` but i am not confident about making that change.

Let me know if that doesn't sound a reasonable explanation for retaining this.